controllers/progressivesync_controller.go

```diff
@@ -334,34 +322,41 @@ func (r *ProgressiveSyncReconciler) updateStatusWithRetry(ctx context.Context, p
 
 // setSyncedAtAnnotation sets the SyncedAt annotation for the currently progressing stage of the app
 func (r *ProgressiveSyncReconciler) setSyncedAtAnnotation(ctx context.Context, app argov1alpha1.Application, stageName string) error {
+
+	log := r.Log.WithValues("stage", stageName)
+
 	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
 		key := client.ObjectKeyFromObject(&app)
 		latest := argov1alpha1.Application{}
+
 		if err := r.Client.Get(ctx, key, &latest); err != nil {
+			log.Info("failed to get app when adding syncedAt annotation")
 			return err
 		}
-		val, ok := app.Annotations[utils.ProgressiveSyncSyncedAtStageKey]
-		// Required due to the use of `omitempty` in serialisation.
-		// `omitempty` converts the empty map into nil.
-		if !ok {
-			r.Log.Info("Initialising empty map for Annotations")
-			latest.Annotations = make(map[string]string)
-		}
-		if val != stageName {
-			r.Log.Info("Setting the annotation for the currently progressing stage")
-			latest.Annotations[utils.ProgressiveSyncSyncedAtStageKey] = stageName
-			if err := r.Client.Update(ctx, &latest); err != nil {
-				return err
+
+		log = log.WithValues("app", fmt.Sprintf("%s/%s", app.Namespace, app.Name))
```

Comment:
could we use latest instead of apps here?

Reply:
Fixed